lib/Analysis/Allocation.cpp

```diff
@@ -135,10 +135,19 @@ getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
   // TODO(jlebar): This is suboptimal if repShape[in/outOrd[0]] is small.  We
   // might be able to merge the few most-minor dimensions and get a larger
   // vector.
-  // For conversions to MmaV1 (Nvidia V100), this inVec is hardcoded in the
-  // codegen.
   inVec = std::min(srcContigPerThread[inOrd[0]], dstContigPerThread[inOrd[0]]);
   outVec = dstContigPerThread[outOrd[0]];
+
+  if (srcLayout.isa<BlockedEncodingAttr>() &&
+      dstLayout.isa<BlockedEncodingAttr>() && outOrd[0] == 0 && inOrd[0] != 0) {
```

Comment:
This isn't really sufficient to detect a transpose, now that transposes can be multidimensional.  For example, we could transpose 1x4x8xf32 -> 1x8x4xf32.

Reply:
I generalized the condition.